cloudsync/runnable.py

```diff
@@ -169,14 +172,21 @@ def start(self, *, daemon=True, **kwargs):
         if self.__shutdown:
             raise RuntimeError("Service was stopped, create a new instance to run.")
         if self.__thread:
-            self.__thread.join(timeout=1)  # give the old thread a chance to die
-        if self.__thread:
-            raise RuntimeError("Service already started")
+            if self.__thread.is_alive():
+                self.__thread.join(timeout=1)  # give the old thread a chance to die
+            if self.__thread.is_alive():
+                raise RuntimeError("Service already started")
         self.__stopping = False
         self.__thread = threading.Thread(target=self.run, kwargs=kwargs, daemon=daemon, name=self.service_name)
         self.__thread.name = self.service_name
         self.__thread.start()
 
+    @property
+    def _thread_id(self):
+        if self.__thread:
+            return self.__thread.ident
+        return None
```

Comment:
i think you don't need this safety check.   just let it raise an exception if there's no thread